arm-datalake-analytics/account/2016-11-01/swagger/account.json

```diff
@@ -1074,37 +1073,26 @@
       "description": "Azure Storage account properties information to update."
     },
     "StorageAccountInfo": {
-      "properties": {
-        "name": {
-          "type": "string",
-          "readOnly": true,
-          "description": "the account name associated with the Azure storage account."
-        },
-        "properties": {
-          "$ref": "#/definitions/StorageAccountProperties",
-          "description": "the properties associated with this storage account.",
-          "x-ms-client-flatten": true
+      "allOf": [
+        {
+          "$ref": "#/definitions/SubResource"
         }
-      },
-      "description": "Azure Storage account information."
-    },
-    "CreateStorageAccountInfo": {
+      ],
       "required": [
-        "properties",
-        "name"
+        "properties"
       ],
       "properties": {
         "name": {
```

Comment:
By doing allOf on Subresource, name will come from there. Here you are redefining the same property again. This is an incorrect swagger spec.
I think this whole thing around making name as a required property of the resource is completely incorrect. Why should name be a required property in the first place. Shouldn't Resource name be coming as a path parameter? Why should the customer pass it again in the body?